Shared/Models/CampusEatery.swift

```diff
@@ -310,6 +310,73 @@ extension CampusEatery {
         return false
     }
 
+    /// A "major" meal of the day: breakfast, lunch, or dinner
+    enum MajorMeal: Int, CaseIterable, CustomStringConvertible, Comparable {
+
+        case breakfast
+        case lunch
+        case dinner
+
+        static func < (lhs: MajorMeal, rhs: MajorMeal) -> Bool {
+            lhs.rawValue < rhs.rawValue
+        }
+
+        init?(hour: Int) {
+            for choice in MajorMeal.allCases {
+                if choice.hours.contains(hour) {
+                    self = choice
+                    return
+                }
+            }
+
+            return nil
+        }
+
+        init(from date: Date) {
+            var calendar = Calendar.current
+            calendar.timeZone = TimeZone(identifier: "America/New_York")!
+
+            let hour = calendar.component(.hour, from: date)
+            self = MajorMeal(hour: hour) ?? .breakfast
+        }
+
+        /// Inverse of MajorMeal.description
+        init?(description: String?) {
+            switch description {
+            case "Breakfast": self = .breakfast
```

Comment:
```suggestion
            case MajorMeal.breakfast.description: self = .breakfast
```

etc.

Another option (which I like less) is to replace the `switch` with a `MajorMeal.allCases.first(where:)`.